### PR TITLE
Adding utility functions

### PR DESCRIPTION
**Pull request by drudenko**

### CODE REVIEW

**Comment by drudenko** [comment]
ping @haozhun @cberner

File: src/main/java/io/airlift/slice/SliceUtf8.java [hunk 1]
```diff
@@ -37,28 +39,45 @@ private SliceUtf8() {}
...
         }
     }
 
     /**
+     * Utility method to identify if the character of a type is a punctuation character
+     * @param type - type of the character
+     * @return true if the character of type is a punctuation character (excluding quote and continuation
+     *  punctuation)
+     */
+    private static boolean isPunctuation(int type) {
```
Comment (haozhun) [inline src/main/java/io/airlift/slice/SliceUtf8.java]: This is a bad name because 3 out of 7 `P` categories (`Pc`, `Pf`, `Pi`) are excluded.

Comment (haozhun) [inline src/main/java/io/airlift/slice/SliceUtf8.java]: Additionally, since Slice is a library. It may be a bad idea to have this here. Talk to @dain

**Comment by haozhun** [comment]
@dain Please make decisions on interfaces.

**Comment by dain** [comment]
I think this function is not generic enough (need by most users) to be in Slice .  For now, I suggest you leave this in Presto.

If other projects end up needing this we'll figure out an API that works for everyone.

**Comment by drudenko** [comment]
We've discussed with @haozhun and thought of changing it in a way: 
- change the name to isWordBreak()
- add quotation punctuation to it. 
Would this be general enough? 

A question, if this doesn't have to go to slice what are my options? Move the code to one of the classes in presto-facebook module?

**Comment by dain** [comment]
For now, lets leave this functionality in Presto.

**Pull request closed without merging**